NitroxClient/GameLogic/InitialSync/CyclopsInitialAsyncProcessor.cs

```diff
@@ -25,16 +25,19 @@ public CyclopsInitialAsyncProcessor(Vehicles vehicles)
         public override IEnumerator Process(InitialPlayerSync packet, WaitScreen.ManualWaitItem waitScreenItem)
         {
             this.waitScreenItem = waitScreenItem;
-            vehicles.VehicleCreated += OnVehicleCreated;
-
             totalCyclopsToLoad = packet.Vehicles.Where(v => v.TechType.Enum() == TechType.Cyclops).Count();
 
-            foreach (VehicleModel vehicle in packet.Vehicles)
+            if (totalCyclopsToLoad > 0)
             {
-                if (vehicle.TechType.Enum() == TechType.Cyclops)
+                vehicles.VehicleCreated += OnVehicleCreated;
+
+                foreach (VehicleModel vehicle in packet.Vehicles)
                 {
-                    Log.Debug($"Trying to spawn {vehicle}");
-                    vehicles.CreateVehicle(vehicle);
+                    if (vehicle.TechType.Enum() == TechType.Cyclops)
```

Comment:
This check is similar to the totalCyclopsToLoad (also checks on enum). It would be best to iterate the packet.Vehicles onces and save it as a cyclops list.